File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/ShoppingCart.cpp
#include "ShoppingCart.h"
#include "ShoppingCart.h"
#include <vector>
#include <alf/library/BasicInputOutput.hpp>
#include <string>
#include <alf/library/collectionfunctions/CollectionFunctions.hpp>
#include "IOrderManagementInterface.h"

namespace Order
{
  ShoppingCart::ShoppingCart()
  {
  }
  void ShoppingCart::addProduct(int productId)
  {
    this->products.push_back(productId);
    alf::library::BasicInputOutput::WriteLine(std::string("Added product ") + std::to_string(productId));
  }
  void ShoppingCart::removeProduct(int productId)
  {
    alf::library::primitivebehaviors::CollectionFunctions::remove(this->products, productId);
    alf::library::BasicInputOutput::WriteLine(std::string("Removed product ") + std::to_string(productId));
  }
  void ShoppingCart::orderProducts()
  {
    for (auto& productId : this->products)
    {
      this->dependency_OrderManagement->addOrder(productId, 1);
    }
    this->dependency_OrderManagement->orderProducts();
    this->products.clear();
    alf::library::BasicInputOutput::WriteLine(std::string("Ordered products"));
  }
}

File: languages/SoftwareComponentLanguage.sandbox/source_gen/SoftwareComponentLanguage/sandbox/Order/ShoppingCart.cs
/// <filename>
///     ShoppingCart.cs
/// </filename>
using System.Collections.Generic;
using Alf.Library;
using Alf.Library.PrimitiveBehaviors;
using Order;

namespace Order
{
    internal class ShoppingCart : Order.IShoppingInterface
    {

        public ShoppingCart()
        {
        }

        public virtual void addProduct(int productId)
        {
            this.products.Add(productId);
            BasicInputOutput.WriteLine("Added product " + IntegerFunctions.ToString(productId));
        }

        public virtual void removeProduct(int productId)
        {
            this.products.Remove(productId);
            BasicInputOutput.WriteLine("Removed product " + IntegerFunctions.ToString(productId));
        }

        public virtual void orderProducts()
        {
            foreach (var productId in this.products)
            {
                this.dependency_OrderManagement.addOrder(productId, 1);
            }
            this.dependency_OrderManagement.orderProducts();
            this.products.Clear();
            BasicInputOutput.WriteLine("Ordered products");
        }

        private System.Collections.Generic.List<int> products = new System.Collections.Generic.List<int>();

        public Order.IOrderManagementInterface dependency_OrderManagement;
    }
}
